src/test/java/io/fusionauth/samlv2/service/DefaultSAMLv2ServiceTest.java

```diff
@@ -1113,6 +1113,40 @@ public void roundTripResponseSignedAssertion(SignatureLocation signatureLocation
     assertEquals(response.assertion.subject.nameIDs.get(0).format, NameIDFormat.EmailAddress.toSAMLFormat());
   }
 
+  @Test(dataProvider = "signatureLocation")
+  public void roundTripResponseFailedRequestSignedAssertion(SignatureLocation signatureLocation, boolean includeKeyInfoInResponse)
+      throws Exception {
+    KeyPairGenerator kpg = KeyPairGenerator.getInstance("RSA");
+    kpg.initialize(2048);
+    KeyPair kp = kpg.generateKeyPair();
+
+    byte[] ba = Files.readAllBytes(Paths.get("src/test/xml/encodedResponse-authnFailed.txt"));
+    String encodedResponse = new String(ba);
+    DefaultSAMLv2Service service = new DefaultSAMLv2Service();
+    AuthenticationResponse response = service.parseResponse(encodedResponse, false, null);
+
+    String encodedXML = service.buildAuthnResponse(response, true, kp.getPrivate(), CertificateTools.fromKeyPair(kp, Algorithm.RS256, "FooBar"), Algorithm.RS256, CanonicalizationMethod.EXCLUSIVE_WITH_COMMENTS, signatureLocation, includeKeyInfoInResponse);
+    response = service.parseResponse(encodedXML, true, new TestKeySelector(kp.getPublic()));
+
+    // Assert the signature is in the correct location based upon the signature option provided.
+    Document document = parseDocument(encodedXML);
+    Node signature = document.getElementsByTagNameNS(XMLSignature.XMLNS, "Signature").item(0);
+    if (signatureLocation == SignatureLocation.Assertion) {
+      // Previous sibling is expected to be the Issuer, and the next sibling should be Subject.
+      assertEquals(signature.getPreviousSibling().getLocalName(), "Issuer");
```

Comment:
Could we assert parent element is `Assertion`?

Reply:
there is no ~spoon~ assertion